jpa/hibernate-transformer/src/main/java/org/jboss/as/hibernate/Hibernate51CompatibilityTransformer.java

```diff
@@ -391,5 +394,17 @@ public void visitInnerClass(String name, String outerName, String innerName, int
         }
     }
 
+    private static int getMajorJavaVersion() {
+        int major = 0;
+        String version = WildFlySecurityManager.getPropertyPrivileged("java.specification.version", null);
+        if (version != null) {
+            if (version.contains(".")) {
+                major = Integer.parseInt(version.split("\\.")[1]);
+            } else {
+                major = Integer.parseInt(version);
+            }
+        }
+        return major;
```

Comment:
Not really that big of a deal, but a group matching seems safer. If for what ever reason this property changes to something like `11.0.0_02` this would return `0`. Below is essentially what is used for determining the version for domains.
```
private static int getMajorJavaVersion() {
    int major = 8;
    String version = WildFlySecurityManager.getPropertyPrivileged("java.specification.version", null);
    if (version != null) {
        Matcher matcher = Pattern.compile("^(?:1\\.)?(\\d+)$").matcher(version);
        if (matcher.find()) {
            major = Integer.valueOf(matcher.group(1));
        }
    }
    return major;
}
```

Reply:
Excellent, I'll push this change soon, thanks!

Reply:
Thanks, your approach works for the test case on jdk11 + jdk8.  Squashed and pushed.